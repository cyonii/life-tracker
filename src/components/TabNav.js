import { Nav } from 'react-bootstrap';

const TabNav = () => (
  <Nav fill variant="pills" justify id="dashboard-nav" className="mt-auto">
    <Nav.Item>
      <Nav.Link eventKey="add-record" className="h-100">
        <i className="bi bi-plus-circle-fill fs-4 mb-1" />
        Add record.
      </Nav.Link>
    </Nav.Item>
    <Nav.Item>
      <Nav.Link eventKey="records" className="h-100">
        <i className="bi bi-graph-up fs-4 mb-1" />
        Track.it
      </Nav.Link>
    </Nav.Item>
    <Nav.Item>
      <Nav.Link eventKey="progress-report" className="h-100">
        <i className="bi bi-pie-chart-fill fs-4 mb-1" />
        Your progress
      </Nav.Link>
    </Nav.Item>
    <Nav.Item>
      <Nav.Link eventKey="more" className="h-100">
        <i className="bi bi-three-dots fs-4 mb-1" />
        More
      </Nav.Link>
    </Nav.Item>
  </Nav>
);

export default TabNav;
