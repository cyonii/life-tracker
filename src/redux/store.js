import { createStore, combineReducers } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import records from './reducers/records';

export default createStore(combineReducers({ records }), composeWithDevTools());
