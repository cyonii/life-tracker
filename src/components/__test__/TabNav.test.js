import { screen, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import TabNav from '../TabNav';

describe('TabNav', () => {
  it('should render', () => {
    render(
      <Provider store={store}>
        <TabNav />
      </Provider>,
    );
    expect(screen.getByTestId('tab-nav')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <TabNav />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
