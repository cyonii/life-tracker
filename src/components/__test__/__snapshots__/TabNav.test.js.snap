// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabNav should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="mt-auto nav nav-pills nav-fill nav-justified"
    data-testid="tab-nav"
    id="dashboard-nav"
  >
    <div
      class="nav-item"
    >
      <a
        class="h-100 nav-link"
        data-rr-ui-event-key="add-record"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-plus-circle-fill fs-4 mb-1"
        />
        Add record.
      </a>
    </div>
    <div
      class="nav-item"
    >
      <a
        class="h-100 nav-link"
        data-rr-ui-event-key="records"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-graph-up fs-4 mb-1"
        />
        Track.it
      </a>
    </div>
    <div
      class="nav-item"
    >
      <a
        class="h-100 nav-link"
        data-rr-ui-event-key="progress-report"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-pie-chart-fill fs-4 mb-1"
        />
        Your progress
      </a>
    </div>
    <div
      class="nav-item"
    >
      <a
        class="h-100 nav-link"
        data-rr-ui-event-key="more"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-three-dots fs-4 mb-1"
        />
        More
      </a>
    </div>
  </div>
</DocumentFragment>
`;
